api/v1alpha1/runner_types.go

```diff
@@ -39,6 +39,8 @@ type RunnerSpec struct {
 	// +optional
 	Containers []corev1.Container `json:"containers,omitempty"`
 	// +optional
+	DockerResources corev1.ResourceRequirements `json:"dockerResources,omitempty"`
```

Comment:
Thanks for the PR! As this is for configuring dockerd rather than docker, I would personally prefer `DockerdReosurces`. WDYT?

Reply:
Sounds good, fixed. Thought also dindResources, but dockerdResources sound better.